Firestore/core/src/remote/remote_store.cc

```diff
@@ -193,7 +193,16 @@ void RemoteStore::SendWatchRequest(const TargetData& target_data) {
   // We need to increment the expected number of pending responses we're due
   // from watch so we wait for the ack to process any messages from this target.
   watch_change_aggregator_->RecordPendingTargetRequest(target_data.target_id());
-  watch_stream_->WatchQuery(target_data);
+
+  if (!target_data.resume_token().empty() ||
+      target_data.snapshot_version() != SnapshotVersion::None()) {
+    int32_t expectedCount =
+        GetRemoteKeysForTarget(target_data.target_id()).size();
+    TargetData new_target_data = target_data.WithExpectedCount(expectedCount);
```

Comment:
Can i update the ocbect referenced by target_data here instead of creating a new TargetData ?

Reply:
focusing only on resumeToken